lib/xapi/problems.rb

```diff
@@ -39,13 +39,25 @@ def problems_in_deprecated_path
 
     def problems
       Dir.glob("%s/metadata/exercises/*/" % root).sort
-         .map { |f| Problem.new(File.basename(f), root) }
+         .map {|f|
+           slug = File.basename(f)
+           Problem.new(slug, root) unless deprecated?(slug)
```

Comment:
I try to avoid having a conditional inside of a loop. For clarity I would rather filter out the deprecated slugs first, and then map over the slugs to create the new problem.